### PR TITLE
meta: use lerna to run scripts

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Move dependencies from packages to root and update ci to use `lerna` to run scripts.

~~Add ts-node and typescript to dev dependencies of @sequelize/core package so that tests can be run locally.~~
~~Without these packages as dev dependencies, running tests locally returns the following error:~~
```
command not found: tsc
```

### CODE REVIEW

**Comment by WikiRik** [comment]
They are already in the `monorepo`, any reason why you only want to do things in the `core` package?

**Comment by lohart13** [comment]
@WikiRik when running the below commands, the errors occur.
```
yarn workspace @sequelize/core test-typings
yarn workspace @sequelize/core teaser
```
These are both used in the command: `yarn workspace @sequelize/core test-sqlite`.

It seems when using `ts-node` or `typescript` at a package level rather than the entire workspace, you need to include the dependencies in the package as well as the workspace.

Running the command `yarn test-typings` for the workspace works perfectly however running this for at either of the packages, the error `command not found: tsc` is produced.

I have updated the PR to include typescript in the dev dependencies for `@sequelize/validator.js` as well.

**Comment by lohart13** [comment]
Also, `concurrently` should be in the root of the workspace since it's used with the `yarn start-all` and `yarn stop-all` commands

**Comment by WikiRik** [comment]
I agree with the `concurrently` change, that should indeed be in the root.
Personally I do not agree that we should include `typescript` and `ts-node` in every package, especially considering that we will have 10+ packages with dialect splitting and multiple validator options in v8. Ín my opinion you should run `yarn` in the root and then you can move on to run scripts for separate packages. But I would like @sequelize/maintainers to pitch in on this as well.

**Comment by lohart13** [comment]
@WikiRik by using `lerna` to run scripts there is no need for `ts-node` and `typescript` as devDependencies in each package. I'll update the PR to remove this.

The [docs for running tests](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#4-running-tests) need updating as some of these commands are no longer valid.

Do you want me to make the changes in this PR or submit another PR to update the this?

Also, I've moved the following devDependencies `cross-env` and `husky` packages to the root dev dependencies.

**Comment by ephys** [comment]
What's weird is that I was running those commands without issues not too long ago. Did something break with the upgrade to yarn modern there? :(

**Comment by lohart13** [comment]
https://github.com/sequelize/sequelize/blob/0338d26680a4393c1eb80188dda22557534389f9/CONTRIBUTING.md?plain=1#L154-L180

Above is the section with the incorrect commands.
@ephys when the repo changed to a mono repo, the commands weren't updated to reflect this, but the by using commands like `yarn workspace @sequelize/core test-sqlite` this worked perfectly until yarn was upgraded to the modern version.

**Comment by WikiRik** [comment]
> What's weird is that I was running those commands without issues not too long ago. Did something break with the upgrade to yarn modern there? :(

They still work, after running `yarn && yarn build` in the root folder. In my view only the moving of `concurrently`, `cross-env` and `husky` to the root package.json is a change we should do, the inclusion of `lerna` in the CI is not needed imo

**Comment by WikiRik** [comment]
> The [docs for running tests](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#4-running-tests) need updating as some of these commands are no longer valid.
> 
> Do you want me to make the changes in this PR or submit another PR to update the this?

I'm fine with either. We probably need to change more in the CONTRIBUTING docs with v7 and moving to a monorepo, but this is already a nice start. Thanks for working on this!

**Comment by lohart13** [comment]
> > What's weird is that I was running those commands without issues not too long ago. Did something break with the upgrade to yarn modern there? :(
> 
> They still work, after running `yarn && yarn build` in the root folder. In my view only the moving of `concurrently`, `cross-env` and `husky` to the root package.json is a change we should do, the inclusion of `lerna` in the CI is not needed imo

Correct, however if we move `cross-env` to the root folder, the integration and unit test will fail in the `@sequelize/core` package that is why I moved the CI to use `lerna`.

**Comment by lohart13** [comment]
> > The [docs for running tests](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#4-running-tests) need updating as some of these commands are no longer valid.
> > Do you want me to make the changes in this PR or submit another PR to update the this?
> 
> I'm fine with either. We probably need to change more in the CONTRIBUTING docs with v7 and moving to a monorepo, but this is already a nice start. Thanks for working on this!

Are you happy to using `lerna` to run the scripts locally?
This is the easier way without having to change any other dependencies.

**Comment by WikiRik** [comment]
I thought about it and I think we'll use these dependencies for scripts in the root package.json in the future more, so I think that using `lerna` is the way to go. Let's also use `lerna` locally and update that in the contributing file. Thanks again for your work on this!

**Comment by lohart13** [comment]
I've updated the contributing doc with some more info around using lerna and replaced the commands with `yarn lerna run` instead of just `yarn`.

LMK if this look's alright.

File: lerna.json
Comment (WikiRik) [inline lerna.json:3]:
```suggestion
  "version": "0.0.0-development"
```

File: CONTRIBUTING.md
Comment (WikiRik) [inline CONTRIBUTING.md:188]: Maybe we should encourage people to go into the package folder before doing this? Since otherwise it will also run the tests of the other packages. (people don't have to use `lerna` in that case, but that's a side-effect)

File: CONTRIBUTING.md
Comment (WikiRik) [inline CONTRIBUTING.md:156]:
```suggestion
Sequelize is a monorepo and uses `lerna` to run scripts in each of the packages. The syntax for the commands is: `yarn lerna run` followed by the script name. For example:
```

File: lerna.json
Comment (lohart13) [reply lerna.json:3]: Both @sequelize/core & @sequelize/validator.js packages have the version set as 7.0.0-alpha.20, is this correct?

File: CONTRIBUTING.md
Comment (lohart13) [reply CONTRIBUTING.md:188]: I think it would make more sense to go to the package folder. I'll update this in the file

File: lerna.json
Comment (WikiRik) [reply lerna.json:3]: I think we'll need to update those as well. Our latest alpha is not 20 so it's better to have something marked as development then an old version imo

File: CONTRIBUTING.md
Comment (lohart13) [reply CONTRIBUTING.md]: Done

File: lerna.json
Comment (lohart13) [reply lerna.json]: Ok, I've reverted the package version to 0.0.0-development as well as lerna

**Comment by lohart13** [comment]
In order to set the version to 0.0.0-development, I had to set @sequelize/validator.js dependency to the workspace version of @sequelize/core.
As per [Yarn's documentation](https://yarnpkg.com/features/workspaces#publishing-workspaces), this value should be substituted for the actual version at time of publish.

File: CONTRIBUTING.md
Comment (lohart13) [reply CONTRIBUTING.md]: Done, LMK what you think

**Review by WikiRik — DISMISSED** [review]
Looks good to me! I'll wait a few days before merging this so @ephys has the chance to check this out as well

File: packages/core/package.json
Comment (ephys) [inline packages/core/package.json:4]: lerna uses the version in package.json to determine the next version. When we merge this on the v7 branch, it is very important to make sure we don't actually replace this version number in the v7 branch.

**Review by ephys — COMMENTED** [review]
> As per [Yarn's documentation](https://yarnpkg.com/features/workspaces#publishing-workspaces), this value should be substituted for the actual version at time of publish.

Publishing is done by lerna. Are we 100% sure it does that?

Based on how lerna 7 works, I think this should be fine, but I really don't want to push a broken build

File: lerna.json
Comment (ephys) [reply lerna.json:3]: Ok the option does not exist anymore

**Review by ephys — APPROVED** [review]

**Pull request merged**
